dht-node/src/dht_node/index.ts

```diff
@@ -190,20 +198,69 @@ async function writeFederatedHomeCommunityEnries(pubKey: any): Promise<Community
   try {
     // first remove privious existing homeCommunity entries
     DbFederatedCommunity.createQueryBuilder().delete().where({ foreign: false }).execute()
-
-    homeApiVersions.forEach(async function (homeApi) {
-      const homeCom = new DbFederatedCommunity()
+    for (let i = 0; i < homeApiVersions.length; i++) {
+      const homeCom = DbFederatedCommunity.create()
       homeCom.foreign = false
-      homeCom.apiVersion = homeApi.api
-      homeCom.endPoint = homeApi.url
-      homeCom.publicKey = pubKey.toString('hex')
-
-      // this will NOT update the updatedAt column, to distingue between a normal update and the last announcement
+      homeCom.apiVersion = homeApiVersions[i].api
+      homeCom.endPoint = homeApiVersions[i].url
+      homeCom.publicKey = Buffer.from(pubKey)
       await DbFederatedCommunity.insert(homeCom)
-      logger.info(`federation home-community inserted successfully: ${JSON.stringify(homeCom)}`)
-    })
+      logger.info(
+        `federation home-community inserted successfully: ${JSON.stringify(homeApiVersions[i])}`,
+      )
+    }
   } catch (err) {
-    throw new Error(`Federation: Error writing HomeCommunity-Entries: ${err}`)
+    throw new Error(`Federation: Error writing federated HomeCommunity-Entries: ${err}`)
   }
   return homeApiVersions
 }
+
+export async function writeHomeCommunityEntry(pubKey: string): Promise<void> {
+  try {
+    // check for existing homeCommunity entry
+    let homeCom = await DbCommunity.findOne({
+      foreign: false,
+      publicKey: Buffer.from(pubKey),
+    })
+    if (!homeCom) {
+      // check if a homecommunity with a different publicKey still exists
+      homeCom = await DbCommunity.findOne({ foreign: false })
+    }
+    if (homeCom) {
+      // simply update the existing entry, but it MUST keep the ID and UUID because of possible relations
+      homeCom.publicKey = Buffer.from(pubKey)
+      homeCom.url = CONFIG.FEDERATION_COMMUNITY_URL + '/api/'
```

Comment:
this URL is incorrect I believe - since we have not clearly defined this its ok for me - but there will be change on this one for sure.

Reply:
As i remember we had several discussions about the url and you just want to handle it as total free text.
But on my opinion and my args in our discussion was always not to handle it as free text. The current implementation in all modules and config is to define an url and the application expect an attached /api/. 
If it is incorrect the previous tests will not work too, but they do.